File: app/src/main/java/it/bitprepared/prbm/mobile/model/entities/EntityOther.java
/*   This file is part of PrbmMobile
 *
 *   PrbmMobile is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   PrbmMobile is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with PrbmMobile.  If not, see <http://www.gnu.org/licenses/>.
 */

package it.bitprepared.prbm.mobile.model.entities;

import it.bitprepared.prbm.mobile.R;
import it.bitprepared.prbm.mobile.model.EntityField;
import it.bitprepared.prbm.mobile.model.EntityFieldType;
import it.bitprepared.prbm.mobile.model.PrbmEntity;

/**
 * Entity used to represent a building
 */
public class EntityOther extends PrbmEntity {

    /** Base View IDs */
    private int id = 1;

    private String type = "Altro";
    private String
        typeDesc = "Utilizza questa classe per inserire informazioni che non rientrano nella classi precedenti.";

    /** Array of Extra fields values */
    private EntityField[] extraFields = {
        new EntityField("Descrizione", "Inserisci la descrizione dell'osservazione",
                        id++, EntityFieldType.LONG_TEXT),
        };

    /**
     * Base constructor for a new PRBM Entity
     * @param description Entity description
     * @param caption     Entity caption
     * @param timestamp   Entity timestamp
     */
    public EntityOther(String description, String caption, String timestamp) {
        super(description, caption, timestamp);
    }

    /**
     * Base constructor for an empty EntityOther
     */
    public EntityOther() {
        this("", "", "");
    }


    /**
     * Base constructor for a new PRBM Entity
     * @param description Entity description
     * @param caption     Entity caption
     * @param timestamp   Entity timestamp
     */
    public EntityOther(String description, String caption, String timestamp, EntityField[] extraFields) {
        this(description, caption, timestamp);
        this.extraFields = extraFields;
    }

    @Override
    public String getType() {
        return type;
    }

    @Override
    public String getTypeDescription() {
        return typeDesc;
    }

    @Override
    public EntityField[] getExtraFields() {
        return this.extraFields;
    }

    @Override
    public int getIdListImage() {
        return R.drawable.background_other_list;
    }

    @Override
    public int getIdButtonImage() {
        return R.drawable.button_other;
    }

    @Override
    public int getIdBackImage() {
        return R.color.other_back;
    }

}

File: app/src/main/java/it/bitprepared/prbm/mobile/model/entities/EntityOther.kt
package it.bitprepared.prbm.mobile.model.entities

import it.bitprepared.prbm.mobile.R
import it.bitprepared.prbm.mobile.model.EntityField
import it.bitprepared.prbm.mobile.model.EntityFieldType
import it.bitprepared.prbm.mobile.model.PrbmEntity

private val defaultExtraField = arrayOf(
    EntityField(
        "Descrizione", "Inserisci la descrizione dell'osservazione", 1, EntityFieldType.LONG_TEXT
    ),
)

/**
 * Entity used to represent a building
 */
class EntityOther @JvmOverloads constructor(
    description: String = "",
    caption: String = "",
    timestamp: String = "",
    override val extraFields: Array<EntityField> = defaultExtraField
) : PrbmEntity(description, caption, timestamp) {

    override val type: String = "Altro"
    override val typeDescription: String =
        "Utilizza questa classe per inserire informazioni che non rientrano nella classi precedenti."

    override val idIconColor: Int
        get() = R.color.other_back
}
